app/models/jellyfish_demo/service/storage.rb

```diff
@@ -0,0 +1,55 @@
+module JellyfishDemo
+  module Service
+    class Storage < ::Service
+      def actions
+        actions = super.merge :terminate
+        actions
+      end
+
+      def provision
+        random_id = 1 + rand(99_999)
+        random_ip = rand(999)
+
+        details = {
+          image_id: product.settings[:image_id],
+          flavor_id: product.settings[:flavor_id],
+          key_name: product.settings[:key_name],
+          subnet_id: product.settings[:subnet_id],
+          instance_id: random_id,
+          public_ip_address: "192.178.0.#{random_ip}"
+        }
+
+        # SAVE PRODUCT DETAILS
+        save_outputs(details, [['image_id', :image_id], ['flavor_id', :flavor_id], ['key_name', :key_name],
```

Comment:
It was copy/pasted from jellyfish_aws.

Should I replace it with `save_outputs(details, ['image_id', 'flavor_id', 'key_name', 'subnet_id', 'instance_id', 'public_ip_address'], ValueTypes::TYPES[:string])`?